mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,462 @@
+
+            self.data_table.add_data(
+                image_name,
+                self.wandb.Image(image, boxes=boxes, classes=self.class_set))
+
+    def _log_predictions(self, results, epoch):
+        table_idxs = self.data_table_ref.get_index()
+        assert len(table_idxs) == self.num_eval_images
+
+        for ndx in table_idxs:
+            result = results[ndx]
+            assert len(result) == len(self.class_id_to_label)
+
+            box_data = []
+            class_scores = []
+            for label_id, bbox_scores in enumerate(result):
+                if len(bbox_scores) != 0:
+                    class_score = 0
+                    count = 0
+                    for bbox_score in bbox_scores:
+                        confidence = float(bbox_score[4])
+                        if confidence > 0.3:
+                            class_score += confidence
+                            count += 1
+                            class_name = self.class_id_to_label[label_id]
+                            box_data.append(
+                                self._get_wandb_bbox(
+                                    bbox_score, label_id,
+                                    f'{class_name} {confidence:.2f}'))
+
+                    class_scores.append(class_score / (count + 1e-6))
+                else:
+                    class_scores.append(0)
+
+            boxes = {
+                'predictions': {
+                    'box_data': box_data,
+                    'class_labels': self.class_id_to_label
+                }
+            }
+
+            self.eval_table.add_data(
+                epoch, self.data_table_ref.data[ndx][0],
+                self.data_table_ref.data[ndx][1],
+                self.wandb.Image(
+                    self.data_table_ref.data[ndx][1],
+                    boxes=boxes,
+                    classes=self.class_set), *tuple(class_scores))
+
+    def _get_wandb_bbox(self, bbox, label, box_caption):
+        """Get structured dict for logging bounding boxes to W&B.
+
+        Args:
+            bbox (list): Bounding box coordinates in
+                        (minX, minY, maxX, maxY) format.
+            label (int): label id for that bounding box.
+            box_caption (str): Caption for that bounding box.
+            scale_factor (list): List rescaling factor for bounding box values.
+
+        Returns:
+            dict: Structured dict required for logging
+                  that bounding box to W&B.
+        """
+        position = dict(
+            minX=int(bbox[0]),
+            minY=int(bbox[1]),
+            maxX=int(bbox[2]),
+            maxY=int(bbox[3]))
+
+        if not isinstance(label, int):
+            label = int(label)
+
+        if not isinstance(box_caption, str):
+            box_caption = str(box_caption)
+
+        return {
+            'position': position,
+            'class_id': label,
+            'box_caption': box_caption,
+            'domain': 'pixel'
+        }
```

Comment:
Can we make this function can directly input bboxes and labels and generate a list of dict. It may make the code simple to read.